beacon_node/network/src/subnet_service/tests/mod.rs

```diff
@@ -213,71 +206,79 @@ mod attestation_service {
         // subscription config
         let committee_index = 1;
         // Keep a low subscription slot so that there are no additional subnet discovery events.
-        let subscription_slot = 0;
-        let committee_count = 1;
         let subnets_per_node = MainnetEthSpec::default_spec().subnets_per_node as usize;
 
         // create the attestation service and subscriptions
-        let mut attestation_service = get_attestation_service(None);
-        let current_slot = attestation_service
+        let mut subnet_service = get_subnet_service();
+        let _events = get_events(&mut subnet_service, None, 1).await;
+
+        let current_slot = subnet_service
             .beacon_chain
             .slot_clock
             .now()
             .expect("Could not get current slot");
 
+        // Generate a subnet that isn't in our permanent subnet collection
+        let subscription_slot = current_slot + 1;
+        let mut committee_count = 1;
+        let mut subnet = Subnet::Attestation(
+            SubnetId::compute_subnet::<MainnetEthSpec>(
+                current_slot,
+                committee_index,
+                committee_count,
+                &subnet_service.beacon_chain.spec,
+            )
+            .unwrap(),
+        );
+        while subnet_service
+            .permanent_subscriptions()
+            .any(|x| *x == subnet)
+        {
+            committee_count += 1;
+            subnet = Subnet::Attestation(
+                SubnetId::compute_subnet::<MainnetEthSpec>(
+                    subscription_slot,
+                    committee_index,
+                    committee_count,
+                    &subnet_service.beacon_chain.spec,
+                )
+                .unwrap(),
+            );
+        }
+
         let subscriptions = vec![get_subscription(
             committee_index,
-            current_slot + Slot::new(subscription_slot),
+            current_slot,
             committee_count,
             true,
         )];
 
         // submit the subscriptions
-        attestation_service
+        subnet_service
             .validator_subscriptions(subscriptions.into_iter())
             .unwrap();
 
         // not enough time for peer discovery, just subscribe, unsubscribe
-        let subnet_id = SubnetId::compute_subnet::<MainnetEthSpec>(
-            current_slot + Slot::new(subscription_slot),
-            committee_index,
-            committee_count,
-            &attestation_service.beacon_chain.spec,
-        )
-        .unwrap();
         let expected = [
-            SubnetServiceMessage::Subscribe(Subnet::Attestation(subnet_id)),
-            SubnetServiceMessage::Unsubscribe(Subnet::Attestation(subnet_id)),
+            SubnetServiceMessage::Subscribe(subnet),
+            SubnetServiceMessage::Unsubscribe(subnet),
         ];
 
         // Wait for 1 slot duration to get the unsubscription event
```

Comment:
Looks like we're waiting for 1 epoch here?

Reply:
The third parameter here is the timeout. It gets the next two events, but if we dont get 2 events in an epoch it times out. 

I think we are waiting for the next events here, which should be the next slot